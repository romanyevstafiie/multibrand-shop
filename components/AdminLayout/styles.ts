import styled from 'styled-components';

export const LayoutContainer = styled.section`
  display: grid;
  grid-template-columns: 1fr 8fr;
  min-height: 100vh;
`;

export const ChildrenWrapper = styled.main`
  background-color: ${({ theme }) => theme.dashboardGrayBg};
`;
