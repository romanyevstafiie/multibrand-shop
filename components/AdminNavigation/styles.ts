import styled from 'styled-components';

export const StyledAdminNavigation = styled.aside`
  display: flex;
  align-items: center;
  justify-content: center;
  position: sticky;
  top: 0;
  padding: 5rem 1rem;

  li {
    margin: 1rem 0;
  }
`;
