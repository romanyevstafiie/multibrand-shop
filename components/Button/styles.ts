import styled, { css, keyframes } from 'styled-components';
import { Props } from './Button';

const defaultButton = css`
  background: ${({ theme }) => theme.primaryGolden};
`;

const hoveredButton = css`
  background: ${({ theme }) => theme.secondaryGolden};
`;

const focusedButton = css`
  box-shadow: 0 0 0 3px ${({ theme }) => theme.secondaryGolden};
`;

const disabledButton = css`
  background: ${({ theme }) => theme.darkGray};
  cursor: not-allowed;
`;

const loadingKeyframes = keyframes`
  0% {
    transform: translateX(25px);
  }
  100% {
    transform: translateX(-20px);
  }
`;

const loadingButton = css`
  &::before {
    content: '';
    position: absolute;
    z-index: 1;
    top: 0;
    left: -100%;
    width: 300%;
    height: 100%;
    background: ${({ theme }) => theme.primaryGolden}
      repeating-linear-gradient(
        60deg,
        transparent,
        transparent 10px,
        #ffa500 10px,
        #ffc130 20px
      );
    animation: ${loadingKeyframes} 1s infinite linear;
  }
  & > span {
    opacity: 0.5;
  }
  cursor: wait;
`;

const completedButton = css`
  padding: 0 2rem;
  background: ${({ theme }) => theme.primaryGreen};
  cursor: default;
`;

export const StyledButton = styled.button<Props>`
  position: relative;
  overflow: hidden;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: 3rem;
  padding: 0 2rem;
  font-family: inherit;
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: 700;
  background: transparent;
  border: 0;
  border-radius: ${({ theme }) => theme.borderRadius};
  color: ${({ theme }) =>
    theme.mode === 'dark' ? theme.primaryWhite : theme.secondaryBg};
  cursor: pointer;
  outline: none;
  & > span {
    position: relative;
    display: flex;
    align-items: center;
    z-index: 1;
  }
  ${(p) => !p.isLoading && defaultButton};
  &:hover {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && hoveredButton};
  }
  &:focus {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && focusedButton};
  }
  &:disabled {
    ${disabledButton};
  }
  ${(p) => p.isLoading && loadingButton};
  ${(p) => p.isCompleted && completedButton};
`;
