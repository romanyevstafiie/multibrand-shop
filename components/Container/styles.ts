import styled from 'styled-components';

export const StyledContainer = styled.section`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: ${({ theme }) => theme.maxWidth};
  margin: 0 auto;
  padding: 0 1rem;

  a {
    text-decoration: none;
  }
`;
