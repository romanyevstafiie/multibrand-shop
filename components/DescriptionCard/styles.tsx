import styled from 'styled-components';

export const CardContainer = styled.article`
  background: ${({ theme }) => theme.secondaryBg};
  border-radius: ${({ theme }) => theme.borderRadius};
  border: 1px solid ${({ theme }) => theme.borderGrayColor};
  padding: 1.25rem;
`;
export const Description = styled.p``;
