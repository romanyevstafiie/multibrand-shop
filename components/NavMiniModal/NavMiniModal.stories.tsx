import { Story } from '@storybook/react';
import React from 'react';
import NavMiniModal, { Props } from './NavMiniModal';

export default {
  title: 'navbar/header/Modal Dropdown',
  component: NavMiniModal,
};

const Template: Story<Props> = (args) => <NavMiniModal {...args} />;

export const Primary = Template.bind({});
const Test = () => <h1>Test</h1>;
Primary.args = { isOpen: true, modalContent: <Test /> };
