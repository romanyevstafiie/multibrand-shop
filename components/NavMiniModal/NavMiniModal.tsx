import React, { FC, ReactNode } from 'react';
import * as Styled from './styles.NavMiniModal';

export interface Props {
  modalContent: ReactNode;
  isOpen: boolean;
}
/**
 * Wrapper for Navbar Dropdowns with User account/cart info
 * @function NavMiniModal
 * @param  {ReactNode} modalContent - dropdown content
 * @param  {boolean} isOpen - dropdown display mode, isOpen = show, otherwise modal will be hidden
 * @returns {number} - Rendered NavMiniModal Component
 */
const NavMiniModal: FC<Props> = ({ modalContent, isOpen }) => {
  return <Styled.Container isOpen={isOpen}>{modalContent}</Styled.Container>;
};
export default NavMiniModal;
