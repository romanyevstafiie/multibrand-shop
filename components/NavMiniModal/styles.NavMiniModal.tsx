import styled from 'styled-components';
import { slideInKeyframes } from '../../styles/reusableStyles';

interface Props {
  isOpen: boolean;
  modalWidth: string;
}

export const Container = styled.div<Props>`
  z-index: 6;
  position: absolute;
  top: 5rem;
  right: 3rem;
  min-width: ${({ modalWidth }) => modalWidth || '250px'};
  border: 1px solid ${({ theme }) => theme.borderGrayColor};
  border-radius: ${({ theme }) => theme.borderRadius};
  background: ${({ theme }) => theme.secondaryBg};
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: 1.25rem;

  animation: ${slideInKeyframes} 0.6s cubic-bezier(0.23, 1, 0.32, 1) both;
`;
