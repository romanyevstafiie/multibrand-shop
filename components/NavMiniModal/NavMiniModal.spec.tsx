import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import { withReduxAndStyledProviders } from '../../test/testUtils';
import NavMiniModal from './NavMiniModal';
import * as Styled from './styles.NavMiniModal';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { mainTheme } from '../../styles/mainTheme';

/**
 * Setup function for the component
 * @returns {setup} shallowWrapper
 */
const setup = (isOpen: boolean) => {
  return mount(
    withReduxAndStyledProviders(
      <NavMiniModal
        isOpen={isOpen}
        modalContent={() => <h1>tomato</h1>}
        top="1rem"
        right="1rem"
        modalWidth="100%"
        animationVariant={'slideInTop'}
      />
    )
  );
};
describe('Navbar Modal Dropdown', () => {
  const wrap = setup(true);

  it('should render component for isOpen state', () => {
    expect(wrap);
    expect(wrap.length).toBe(1);
  });

  it('should use specified props', () => {
    const modal = wrap.find(Styled.Container);
    expect(modal.prop('top')).toBe('1rem');
    expect(modal.prop('right')).toBe('1rem');
  });
});

describe('it apply specified styles for open/close states', () => {
  test('it applies default styles for open modal', () => {
    const tree = renderer
      .create(<ThemeProvider theme={mainTheme}>{setup(true)}</ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('visibility', 'visible');
  });

  test('it does not shoul modal when it is closed', () => {
    const tree = renderer
      .create(<ThemeProvider theme={mainTheme}>{setup(false)}</ThemeProvider>)
      .toJSON();

    expect(tree).toHaveStyleRule('visibility', 'hidden');
  });
});
