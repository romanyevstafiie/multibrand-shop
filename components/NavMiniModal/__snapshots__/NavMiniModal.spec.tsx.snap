// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it apply specified styles for open/close states it applies default styles 1`] = `
.c0 {
  visibility: visible;
  opacity: 1;
  z-index: 6;
  position: absolute;
  top: 1rem;
  right: 1rem;
  min-width: 100%;
  max-height: 90vh;
  overflow-y: auto;
  border: 1px solid rgba(143,143,143,0.1);
  border-radius: 0.5rem;
  background: #151515;
  padding: 1.25rem;
  -webkit-animation: dfiLNz 0.5s cubic-bezier(0.25,0.46,0.45,0.94) both;
  animation: dfiLNz 0.5s cubic-bezier(0.25,0.46,0.45,0.94) both;
}

@media (max-width:768px) {
  .c0 {
    padding: 0.75rem;
  }
}

<div
  className="c0"
/>
`;

exports[`it apply specified styles for open/close states it applies default styles for open modal 1`] = `
.c0 {
  visibility: visible;
  opacity: 1;
  z-index: 6;
  position: absolute;
  top: 1rem;
  right: 1rem;
  min-width: 100%;
  max-height: 90vh;
  overflow-y: auto;
  border: 1px solid rgba(143,143,143,0.1);
  border-radius: 0.5rem;
  background: #151515;
  padding: 1.25rem;
  -webkit-animation: dfiLNz 0.5s cubic-bezier(0.25,0.46,0.45,0.94) both;
  animation: dfiLNz 0.5s cubic-bezier(0.25,0.46,0.45,0.94) both;
}

@media (max-width:768px) {
  .c0 {
    padding: 0.75rem;
  }
}

<div
  className="c0"
/>
`;
